torch/_dynamo/variables/builtin.py

```diff
@@ -1216,13 +1217,58 @@ def call_setattr(
             tx.output.side_effects.is_attribute_mutation(obj)
             and name_var.is_python_constant()
         ):
-            name = name_var.as_python_constant()
-            if name == "requires_grad" and isinstance(obj, variables.TensorVariable):
-                unimplemented(
-                    "mutating requires_grad can introduce a new leaf from non-leaf or vice versa in "
-                    "the middle of the graph, which aot_autograd does not currently know how to handle. "
-                )
-            tx.output.side_effects.store_attr(obj, name, val)
+            if isinstance(obj, variables.TensorVariable):
+                from .builder import wrap_fx_proxy
+
+                name = name_var.as_python_constant()
+                if name == "requires_grad" and isinstance(
+                    obj, variables.TensorVariable
+                ):
+                    # TODO(voz): Make it work properly
```

Comment:
do we need this to work for FSDP? Or is graph break an okay long term solution here

Reply:
Well this is setattr only, and this was disabled in https://github.com/pytorch/pytorch/pull/113163 - so I am unsure. In this PR, this is just code motion.